javascript/lib/api/src/model/model.ts

```diff
@@ -103,9 +98,9 @@ export abstract class IndexedEntityModel<T extends IndexedEntityModel<T, V>, V>
    * @param mapValue - how to get the value from an objects value.
    * @param mapKey - how to get the key from an objects key.
    */
-  static fromPlainObject<T>(objectToMap: Object | undefined,
-                            mapValue: (value: any, key: string) => T,
-                            mapKey: (key: string, value: any) => string = key => key): IndexedEntityModelType<T> | undefined {
+  static fromPlainObject<T extends EntityModel>(objectToMap: Object | undefined,
```

Comment:
I think it would be good to always return a valid object (`{}`) here instead of falling back to `undefined`. This way we wouldn't need to implement the `undefined` check in every implementing class.

Reply:
Good idea, thanks!